source/CILobbyMenu.cpp

```diff
@@ -228,24 +219,26 @@ void LobbyMenu::update(MenuState& state, string& joingame, string& playername, s
         case MenuState::MainToLobby:
         {
             // handle displaying for Host
-            joingame.clear();
+            _gameSettings->setGameId("");
+
             _nextState = state; // prep to setDisplay
             setDisplay(true);
-            _lobbyRoomLabel->setText(joingame);
-            _gamelobbyplayerlabel1->setText(playername);
+
+            _lobbyRoomLabel->setText(_gameSettings->getGameId());
+            _gamelobbyplayerlabel1->setText(_playerSettings->getPlayerName());
 
             // handle game lobby settings (only visible to the host)
-            _currSpawn = _currGrav = _currWin = 2;
-            _currColor = 2;
-            _lobbySpawnRate = spawnRate = _spawnRates[_currSpawn];
-            _lobbyGravityStrength = gravStrength = _gravStrengths[_currGrav];
-            _lobbyColorCount = colorCount = _colorCounts[_currColor];
-            _lobbyWinPlanetMass = winPlanetMass = _winMass[_currWin];
-
-            _spawnRateLabel->setText(cugl::strtool::to_string(_lobbySpawnRate, 1) + "X");
-            _gravStrengthLabel->setText(cugl::strtool::to_string(_lobbyGravityStrength, 1) + "X");
-            _colorCountLabel->setText(cugl::strtool::to_string(_lobbyColorCount));
-            _winMassLabel->setText(cugl::strtool::to_string(_lobbyWinPlanetMass));
+            _currSpawn = _currGrav = _currWin = _currColor = 2;
+
+            _gameSettings->setSpawnRate(_spawnRates[_currSpawn]);
```

Comment:
it looks like your setting here and in the listeners. I think you should only need to set in the listeners